eureka/src/main/java/com/linecorp/armeria/internal/common/eureka/InstanceInfo.java

```diff
@@ -0,0 +1,406 @@
+                        @JsonProperty("leaseInfo") LeaseInfo leaseInfo,
+                        @Nullable @JsonProperty("metadata") Map<String, String> metadata) {
+        this.instanceId = instanceId;
+        this.hostName = hostName;
+        this.appName = appName;
+        this.appGroupName = appGroupName;
+        this.ipAddr = ipAddr;
+        this.vipAddress = vipAddress;
+        this.secureVipAddress = secureVipAddress;
+        this.port = requireNonNull(port, "port");
+        this.securePort = requireNonNull(securePort, "securePort");
+        this.status = requireNonNull(status, "status");
+        this.homePageUrl = homePageUrl;
+        this.statusPageUrl = statusPageUrl;
+        this.healthCheckUrl = healthCheckUrl;
+        this.secureHealthCheckUrl = secureHealthCheckUrl;
+        this.dataCenterInfo = dataCenterInfo;
+        this.leaseInfo = requireNonNull(leaseInfo, "leaseInfo");
+        if (metadata != null) {
+            this.metadata = metadata;
+        } else {
+            this.metadata = ImmutableMap.of();
+        }
+
+        lastUpdatedTimestamp = System.currentTimeMillis();
+        lastDirtyTimestamp = lastUpdatedTimestamp;
+    }
+
+    /**
+     * Returns the ID of this instance.
+     */
+    @Nullable
+    public String getInstanceId() {
+        return instanceId;
+    }
+
+    /**
+     * Return the name of the application which this instance belongs to.
+     */
+    @Nullable
+    @JsonProperty("app")
+    public String getAppName() {
+        return appName;
+    }
+
+    /**
+     * Return the group name of the application which this instance belongs to.
+     */
+    @Nullable
+    public String getAppGroupName() {
+        return appGroupName;
+    }
+
+    /**
+     * Return the hostname of this instance.
+     */
+    @Nullable
+    public String getHostName() {
+        return hostName;
+    }
+
+    /**
+     * Returns the IP address of this instance.
+     */
+    @Nullable
+    public String getIpAddr() {
+        return ipAddr;
+    }
+
+    /**
+     * Returns the VIP address of this instance.
+     */
+    @Nullable
+    public String getVipAddress() {
+        return vipAddress;
+    }
+
+    /**
+     * Returns the secure VIP address of this instance.
+     */
+    @Nullable
+    public String getSecureVipAddress() {
+        return secureVipAddress;
+    }
+
+    /**
+     * Returns the {@link PortWrapper} of this instance.
+     */
+    public PortWrapper getPort() {
+        return port;
+    }
+
+    /**
+     * Returns the secure {@link PortWrapper} of this instance.
+     */
+    public PortWrapper getSecurePort() {
+        return securePort;
+    }
+
+    /**
+     * Returns the {@link InstanceStatus} of this instance.
+     */
+    public InstanceStatus getStatus() {
+        return status;
+    }
+
+    /**
+     * Returns the home page URL of this instance.
+     */
+    @Nullable
+    public String getHomePageUrl() {
+        return homePageUrl;
+    }
```

Comment:
Do we need this information?  What do you think of ignoring this field from the JSON response?

Reply:
The clients who are using other implementation (not `EurekaEndpointGroup`) might need this information.